# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

This patch adds a preprocessor flag that removes the dependency on the non-open
source OpenAccess library. This ought to be upstreamed, at least in some form.
(I think it would be appropriate to make no-OpenAccess the default.)

diff --git src/pybag/core.cpp src/pybag/core.cpp
index 38a6097..a6426cd 100644
--- src/pybag/core.cpp
+++ src/pybag/core.cpp
@@ -53,6 +53,7 @@ limitations under the License.
 #include <pybag/logging.h>
 #include <pybag/name.h>
 #include <pybag/oa.h>
+#include <pybag/schematic.h>
 #include <pybag/rtree.h>
 #include <pybag/tech.h>
 #include <pybag/transform.h>
@@ -74,8 +75,12 @@ PYBIND11_MODULE(core, m) {
     bind_layout(m);
     // make sure tech/routing grid are defined before binding gds
     bind_gds(m);
+#ifdef PYBAG_OPEN_SOURCE
+    bind_schematic(m);
+#else
     // make sure routing grid and tr_colors are defined before binding OA
     bind_oa(m);
+#endif
     bind_rtree(m);
     bind_logging(m);
     bind_name(m);
